java/ql/src/semmle/code/java/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1728,37 +1732,42 @@ private class PathNodeSink extends PathNode, TPathNodeSink {
  * a callable is recorded by `cc`.
  */
 private predicate pathStep(PathNodeMid mid, Node node, CallContext cc, AccessPath ap) {
```

Comment:
The pruning introduced here in `pathStep` should be mirrored in `partialPathStep`.

Reply:
Mirroring the whole pruning is difficult, as partialPathStep uses the smallstep localflow relation, not the bigstep one.
Pruning by using local call contexts is right now only done for the bigstep relation.
Would it be fine by you to just prune the interprocedural steps in partialPathStep, and keep the local flow as-is?

Reply:
No, it's actually simpler to add pruning to `partialPathStep` as you won't need to mess around with locally relevant call contexts.